### ISSUE TITLE
Invalid service_data, no brightness or color attributes found for *some* hue lights

### ISSUE THREAD

**Issue by Greminn** [basnijholt/adaptive-lighting#661]
# Home Assistant Adaptive Lighting Issue Template

## Bug Reports

If you need help with using or configuring Adaptive Lighting, please [open a Q&A discussion thread here](https://github.com/basnijholt/adaptive-lighting/discussions/new?category=q-a) instead.

### Before submitting a bug report, please follow these troubleshooting steps:

Please confirm that you have completed the following steps:

- [X] I have updated to the [latest Adaptive Lighting version](https://github.com/basnijholt/adaptive-lighting/releases) available in [HACS](https://hacs.xyz/).
- [X] I have reviewed the [Troubleshooting Section](https://github.com/basnijholt/adaptive-lighting#troubleshooting) in the [README](https://github.com/basnijholt/adaptive-lighting#readme).
- [X] (If using Zigbee2MQTT) I have read the [Zigbee2MQTT troubleshooting guide](https://github.com/basnijholt/adaptive-lighting#zigbee2mqtt) in the [README](https://github.com/basnijholt/adaptive-lighting#readme).
- [X] I have checked the [V2 Roadmap](https://github.com/basnijholt/adaptive-lighting/discussions/291) and [open issues](https://github.com/basnijholt/adaptive-lighting/issues) to ensure my issue isn't a duplicate.

### Required information for bug reports:

Please include the following information in your issue.

*Issues missing this information may not be addressed.*

1.  **Debug logs** captured while the issue occurred. [See here for instructions on enabling debug logging](https://github.com/basnijholt/adaptive-lighting#troubleshooting):

```
2023-07-26 08:47:27.849 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140180791355216] Invalid service_data, no brightness or color attributes found: service_data={'entity_id': 'light.lounge_floor_lamp', 'transition': 1.0}
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 205, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1965, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2005, in _execute_service
    return await cast(
           ^^^^^^^^^^^
  File "/config/custom_components/adaptive_lighting/hass_utils.py", line 46, in service_func_proxy
    await intercept_func(call, data)
  File "/config/custom_components/adaptive_lighting/switch.py", line 1965, in _service_interceptor_turn_on_handler
    adaptation_data = await adaptive_switch.prepare_adaptation_data(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/adaptive_lighting/switch.py", line 1192, in prepare_adaptation_data
    return prepare_adaptation_data(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/adaptive_lighting/adaptation_utils.py", line 200, in prepare_adaptation_data
    lighting_type = _identify_lighting_type(service_data)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/adaptive_lighting/adaptation_utils.py", line 165, in _identify_lighting_type
    raise NoColorOrBrightnessInServiceDataError(msg)
custom_components.adaptive_lighting.adaptation_utils.NoColorOrBrightnessInServiceDataError: Invalid service_data, no brightness or color attributes found: service_data={'entity_id': 'light.lounge_floor_lamp', 'transition': 1.0}
2023-07-26 08:48:04.616 WARNING (MainThread) [aiohomekit.controller.coap.pdu] Transaction 0 failed with error 6 (Invalid request
2023-07-26 08:48:52.219 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/adaptive_lighting/switch.py", line 1114, in _async_update_at_interval_action
    await self._update_attrs_and_maybe_adapt_lights(
  File "/config/custom_components/adaptive_lighting/switch.py", line 1399, in _update_attrs_and_maybe_adapt_lights
    await self._adapt_light(light, transition, context=context)
  File "/config/custom_components/adaptive_lighting/switch.py", line 1225, in _adapt_light
    data = await self.prepare_adaptation_data(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/adaptive_lighting/switch.py", line 1192, in prepare_adaptation_data
    return prepare_adaptation_data(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/adaptive_lighting/adaptation_utils.py", line 200, in prepare_adaptation_data
    lighting_type = _identify_lighting_type(service_data)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/adaptive_lighting/adaptation_utils.py", line 165, in _identify_lighting_type
    raise NoColorOrBrightnessInServiceDataError(msg)
custom_components.adaptive_lighting.adaptation_utils.NoColorOrBrightnessInServiceDataError: Invalid service_data, no brightness or color attributes found: service_data={'entity_id': 'light.office_light_1', 'transition': 45.0}
```

2.  [Your Adaptive Lighting configuration](https://github.com/basnijholt/adaptive-lighting#gear-configuration):

```
Sorry I dont know how to export the config
```

3.  (If using Zigbee2MQTT), provide your configuration files (**remove all personal information before posting**):
    - `devices.yaml`
    - `groups.yaml`
    - `configuration.yaml` ⚠️; **Warning** _**REMOVE ALL of the PERSONAL INFORMATION BELOW before posting**_ ⚠️;
      - mqtt: `server`:
      - mqtt: `user`:
      - mqtt: `password`:
      - advanced: `pan_id`:
      - advanced: `network_key`:
      - anything in `log_syslog` if you use this
    - Brand and model number of problematic light(s)
```

```

4.  Describe the bug and how to reproduce it:

Enable Adaptive Lighting, issue happens, disable lights work as expected

5. Steps to reproduce the behavior:

**Comment by Greminn** [comment]
It only seems to happen for some hue lights which is strange (note all of my hue light bulbs are just the white/warm white kind), I have not worked out why this is. My hue light strips and nanoleaf light strips are fine.

**Comment by basnijholt** [comment]
Hi @Greminn, thanks a lot for reporting!

It would be really valuable to have the debugging logs because from just that error message alone I cannot figure out what happened.

Check https://github.com/basnijholt/adaptive-lighting#sos-troubleshooting

**Comment by basnijholt** [comment]
I am relatively sure that #662 fixes this issue, I released v1.17.2.

Let me know if this works.

**Comment by Greminn** [comment]
> I am relatively sure that #662 fixes this issue, I released v1.17.2.
> 
> Let me know if this works.

Works ✅ thank you
